src/filtering/markov_kernel.jl

```diff
@@ -216,7 +238,7 @@ function compute_backward_kernel!(
     x::SRGaussian{T,<:IsometricKroneckerProduct},
     K::KT2;
     C_DxD::AbstractMatrix,
-    diffusion=1,
+    diffusion::Union{Number,Diagonal}=1,
```

Comment:
We should have more tests for the different diffusions in the filtering tests I think